assets/tests/QuestionSchemaValidationServiceSpec.js

```diff
@@ -47,6 +47,18 @@ describe('QuestionSchemaValidationService', function() {
     });
   }));
 
+  describe('validateTitlesAreUnique', function() {
+    it('should verify that all questions have a unique title', function() {
+      var titles = new Set();
+      expect(questions.every(function(question) {
+        var title = question.getTitle();
+        var isUnique = !(title in titles);
+        titles.add(title);
+        return isUnique;
+      })).toBe(true);
```

Comment:
Well, yeah, but then every reader does have to do a bit of thinking to make sure it works right, and I kinda think you want to avoid that where possible :)